flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateTest.java

```diff
@@ -499,20 +504,23 @@ public void testUpdateUnknownInputChannel() throws Exception {
 			assertThat(inputGate.getInputChannels().get(localResultPartitionId.getPartitionId()),
 				is(instanceOf((UnknownInputChannel.class))));
 
+			ResourceID localLocation = ResourceID.generate();
+
 			// Trigger updates to remote input channel from unknown input channel
-			inputGate.updateInputChannel(new InputChannelDeploymentDescriptor(
-				remoteResultPartitionId,
-				ResultPartitionLocation.createRemote(InputChannelBuilder.STUB_CONNECTION_ID)));
+			ShuffleDescriptor remoteSdd = createRemoteWithIdAndLocation(
+				remoteResultPartitionId.getPartitionId(),
+				ResourceID.generate());
+			inputGate.updateInputChannel(localLocation, new PartitionInfo(new IntermediateDataSetID(), remoteSdd));
```

Comment:
could reuse createPartitionInfo(remoteResultPartitionId, ResourceID.generate()) to create PartitionInfo here.